src/language/records.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Records
+description: Summary of the record data structure in Dart.
+toc: false
+---
+
+Records are an anonymous, immutable, aggregate type. Like other [collection types][], 
+they let you bundle multiple objects into a single object. Unlike other collection 
+types, records are fixed-sized, heterogeneous, and typed.
+
+Records are real values; you can store them in variables, pass them to and from
+functions, and store them in lists.
+
+## Record syntax
+
+Records are comma-delimited lists of named or positional fields,
+enclosed in parenthesis:
+
+```dart
+var record = ('first', a: 2, b: true, 'last');    // Record expression
+```
+
+Named fields inside record type annotations go inside a brace-delimited section
+of type and name pairs:
+
+```dart
+(Sring, {int a, bool b}, String?) record;    // Record type annotation
+```
+Naming fields in record type annotations is optional for documentation purposes.
+The names don't need to match any record expression corresponding to the type.
+
+## Record fields
+
+Record fields are accessible through built-in getters. Records are immutable,
+so fields do not have setters. 
+
+Named fields expose getters of the same name. Positional fields expose getters
+of the name `$<position>`, skipping named fields:
+
+```dart
+print(record.$1);    // Prints '123'
+print(record.a);     // Prints 2
+print(record.b);     // Prints true
+print(record.$2);    // Prints 'last'
+```
+
+To streamline record field access even more, see the page on [Patterns][].
+
+{% comment %}
+    TODO: link to patterns page, specifically records destructuring section.
+{% endcomment %}
+
+## Record types
+
+There is no explicit record type; records are
```

Comment:
It's probably more accurate to say that there are a plethora of record types, one for each unique set of field names, positions, and field types. So `(int, int)` is a record type, `({x: int, y: int})` is another record type, etc.

Also, all of those record types inherit from a built in type named `Record`. So there sort of is "an" explicit record type, though `Record` isn't useful for much. (It's directly analogous to `Function` and function types.)

I would maybe word this like:

> Records are structurally typed: the set of the fields, their types, and their names, uniquely determines the type of a record. Two records with the same set of field names and types have the same type.

Reply:
Is it accurate to say a records shape is "the the set of its fields, the fields' types, and their names (if any)"? I use "shape" later on in the page so it seems like this would be a good place to define it first